ibm/service/project/resource_ibm_project_environment.go

```diff
@@ -0,0 +1,550 @@
+// Copyright IBM Corp. 2023 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package project
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/project-go-sdk/projectv1"
+)
+
+func ResourceIbmProjectEnvironment() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIbmProjectEnvironmentCreate,
+		ReadContext:   resourceIbmProjectEnvironmentRead,
+		UpdateContext: resourceIbmProjectEnvironmentUpdate,
+		DeleteContext: resourceIbmProjectEnvironmentDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"project_id": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.InvokeValidator("ibm_project_environment", "project_id"),
+				Description:  "The unique project ID.",
+			},
+			"definition": &schema.Schema{
+				Type:        schema.TypeList,
+				MinItems:    1,
+				MaxItems:    1,
+				Required:    true,
+				Description: "The environment definition.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name of the environment.  It is unique within the account across projects and regions.",
+						},
+						"description": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The description of the environment.",
+						},
+						"authorizations": &schema.Schema{
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: "The authorization details. You can authorize by using a trusted profile or an API key in Secrets Manager.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"trusted_profile_id": &schema.Schema{
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "The trusted profile ID.",
+									},
+									"method": &schema.Schema{
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "The authorization method. You can authorize by using a trusted profile or an API key in Secrets Manager.",
+									},
+									"api_key": &schema.Schema{
+										Type:        schema.TypeString,
```

Comment:
This is sensitive can we add Sesitive : true

Reply:
@hkantare ok, I've regenerated the code by setting "x-terraform-sensitive: true" in the swagger.yaml